### PR TITLE
Toppview refactor

### PR DESCRIPTION
**Pull request by cbielow**
some refactoring to make TOPPViewBase a bit more managable...

reduces line count of TOPPViewBase from 4400 to 3700 lines...
Some more refactoring will be required to make the class readable again...

### CODE REVIEW

File: src/openms/include/OpenMS/KERNEL/MSExperiment.h [hunk 1]
```diff
@@ -545,6 +545,13 @@ namespace OpenMS
     */
     void clear(bool clear_meta_data);
 
+    /// returns true if at least one of the spectra has level==1
+    bool containsMS1Scans() const;
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/MSExperiment.h:549]: I would be fine with having this in MSExperiment. But maybe more generic by passing the level?

File: src/openms/include/OpenMS/KERNEL/MSExperiment.h [hunk 2]
```diff
@@ -545,6 +545,13 @@ namespace OpenMS
...
+
+    /// returns true if any MS1 spectra contain at least one peak with intensity of 0.0
+    bool hasMS1Zeros() const;
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/MSExperiment.h:552]: same here

File: src/openms/source/KERNEL/MSSpectrum.cpp [hunk 3]
```diff
@@ -699,4 +699,17 @@ namespace OpenMS
   bool MSSpectrum::RTLess::operator()(const MSSpectrum &a, const MSSpectrum &b) const {
     return a.getRT() < b.getRT();
   }
+
+  // static
+  bool MSSpectrum::containsIMData() const
```
Comment (timosachsenberg) [inline src/openms/source/KERNEL/MSSpectrum.cpp:704]: maybe better to return bool and type?

File: src/openms_gui/include/OpenMS/VISUAL/FilterList.h [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  class TOPPViewBase;
+
+  namespace Internal
+  {
+    /**
+      @brief A widget shows a list of input files (i.e. existing files on a mounted drive),
+             which allows adding/removing files and supports drag'n'drop from the window manager.
+
+    */
+    class FilterList : public QWidget
```
Comment (timosachsenberg) [inline src/openms_gui/include/OpenMS/VISUAL/FilterList.h:61]: much cleaner. nice

File: src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp [hunk 5]
```diff
@@ -262,13 +263,11 @@ namespace OpenMS
     menuBar()->addMenu(help);
     help->addAction(QWhatsThis::createAction(help));
     help->addSeparator();
-    QAction* action = help->addAction("OpenMS website", this, SLOT(showURL()));
-    action->setData("http://www.OpenMS.de");
-    action = help->addAction("Tutorials and documentation", this, SLOT(showURL()), Qt::Key_F1);
-    action->setData(String("html/index.html").toQString());
+    help->addAction("OpenMS website", [&]() { GUIHelpers::openURL("http://www.OpenMS.de"); });
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp:266]: so much cleaner 👍

**Comment by timosachsenberg** [comment]
nice

File: src/openms/source/KERNEL/MSSpectrum.cpp (hunk 3 above)
Comment (cbielow) [reply src/openms/source/KERNEL/MSSpectrum.cpp:703]: what type?

Comment (timosachsenberg) [reply src/openms/source/KERNEL/MSSpectrum.cpp:703]: I thought type of ion mobility that is stored... maybe some CVTerm ... but not urgent

Comment (cbielow) [reply src/openms/source/KERNEL/MSSpectrum.cpp:703]: I'm not sure this is well defined. Will omit that for now

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/viewBuildError.php?buildid=2256

**Pull request merged**